swephp.c

```diff
@@ -2119,43 +2119,56 @@ Converts julian day number to calendar date
 =head3 Parameters
 
   double	jd		julian day number
-  int		gregflag	SE_GREG_CAL (==1) or SE_JUL_CAL (==0)
+  int		gregflag	SE_GREG_CAL (==1, default) or SE_JUL_CAL (==0)
 
 =head3 return array
 
   ['year']		int
   ['month']		int
   ['day']		int
   ['hour']		double
+  ['ihour']		int
+  ['imin']		int
+  ['isec']		int
+  ['dsec']		double
 
 =head3 C declaration
 
   void swe_revjul ( double jd, int gregflag, int *jyear, int *jmon, int *jday, double *jut);
 
-
 =cut
  }}} */
 PHP_FUNCTION(swe_revjul)
 {
-	int year, month, day;
-	long gregflag;
-	double hour, jd;
+	int year, month, day, ihour, imin, isec;
+	long gregflag = 1;
+	double hour, jd, dmin, dsec;
 	
-	if(ZEND_NUM_ARGS() != 2) WRONG_PARAM_COUNT;
+	if (ZEND_NUM_ARGS() < 1 || ZEND_NUM_ARGS() > 2) WRONG_PARAM_COUNT;
 
-	if (zend_parse_parameters(ZEND_NUM_ARGS(), "dl",
+	if (zend_parse_parameters(ZEND_NUM_ARGS(), "d|l",
 			&jd, &gregflag) == FAILURE) {
 		return;
 	}
 
 	swe_revjul(jd, gregflag, &year, &month, &day, &hour);
 
+    ihour = floor(hour);
+    dmin = (hour - ihour) * 60;
+    imin = floor(dmin);
+    dsec = (dmin - imin) * 60;
+    isec = floor(dsec);
```

Comment:
In perl-sweph, I use this code, where I round to 0.05 milliseconds
probably to avoid unexpected results, when swe_julday and swe_revjul
are used in sequence. One does not want 6:59:59.9999 returned when one
entered hour=7.00


  dsec = dhou * 3600.0 + 0.00005;
  ihou = (int) (dsec / 3600.0);
  dsec -= ihou * 3600.0;
  imin = (int) (dsec / 60.0);
  dsec -= imin * 60.0;
  isec = (int) dsec;